### DISCUSSION TITLE
optional form parameters default to value "string" in the ui

### DISCUSSION THREAD

**Discussion by gabor-lbl** [fastapi/fastapi#11852] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post(
    "/jobs/{machine:str}", 
    response_model=models.CommandOutput, 
)
def submit_job(
    job : str = Form(...),
    callbackEmail : Optional[str] = Form(None),
```

### Description

Starting with fastapi 0.111.1, in the swagger ui the form element for "callbackEmail" contains the value "string". In 0.111.0 the form element is empty and the "Send empty value" checkbox is checked. Thanks for any info on this!

### Operating System

macOS

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

Python 3.10.2

**Comment by YuriiMotov** [comment, marked as answer]
The behavior has been changed (because FastAPI now uses the latest version of Swagger-ui possible), but there is nothing wrong here.
You can remove the "string" value and "Send empty value" will be checked

**Reply by gabor-lbl** [reply]
Thank you for your response! Is it possible to do this declaratively from code? I would prefer users to not have to edit optional fields.

**Reply by YuriiMotov** [reply]
I don't now the way how to do this declaratively. I would ask this question in Swagger-UI repository

**Reply by gabor-lbl** [reply]
Ok thanks, I will try there.

**Reply by WaterKnight1998** [reply]
@gabor-lbl did you find anything??

**Reply by gabor-lbl** [reply]
No, we've pinned v0.111.0 for now.

**Comment by ragaman555** [comment]
I also stumbled across the issue.
We can no longer use some endpoints with many body params effectively via the swagger ui, as the nonsense default values has to be deleted for all of them so that an empty value is transferred.

This has been the case since the swagger ui was updated to 5.13.
I have created a bug report there - https://github.com/swagger-api/swagger-ui/issues/10352

Then the problem could possibly be solved by transmitting a null default value.

**Comment by harichallaforu-bit** [comment]
for optional field i am getting error when passing empty to float filed after updating latest version

fastapi==0.110.2 with this it was working fine after updating to fastapi==0.115.2 giving error 

any suggestion please

**Reply by YuriiMotov** [reply]
Use `BeforeValidator` to convert empty string to `None`

**Reply by harichallaforu-bit** [reply]
it's not working 

`
    from typing import Optional
    from pydantic import BaseModel, StrictFloat, Field, field_validator
    
    class TemperatureRequest(BaseModel):
            temperature: Optional[StrictFloat] = Form(None, description="Temperature value, optional")
            
    @field_validator("temperature", mode="before")
    def empty_string_to_none(cls, v):
        """
        Convert empty string from form input to None
        """
        if v in ("", None):
            return None
        return v

    @classmethod
    def as_form(cls, temperature: Optional[float] = Form(None)):
        """
        Form input version (FastAPI Form sends string)
        """
        return cls(temperature=temperature)
        
        
        
    # API code
         
      from fastapi import FastAPI, Form, Depends
      from models import TemperatureRequest
      
      app = FastAPI()
      
      @app.post("/temperature")
      async def post_temperature(
          data: TemperatureRequest = Depends(TemperatureRequest.as_form)
      ):
          return {"temperature": data.temperature} 
        
        
        
`




Before validator is not working here has Optional checking empty value 

Thank you

**Reply by YuriiMotov** [reply]
This works:
```py
from typing import Optional
from pydantic import BaseModel, StrictFloat, field_validator
from fastapi import FastAPI, Form

class TemperatureRequest(BaseModel):
    temperature: Optional[StrictFloat] = Form(None, description="Temperature value, optional")

    @field_validator("temperature", mode="before")
    def empty_string_to_none(cls, v):
        """
        Convert empty string from form input to None
        """
        if v in ("", None):
            return None
        return v

app = FastAPI()

@app.post("/temperature")
async def post_temperature(
    data: TemperatureRequest = Form(TemperatureRequest)
):
    return {"temperature": data.temperature} 

```
